### PR TITLE
Fix formatting JSON for ivars with objects, and format JSON full body responses

### PR DESCRIPTION
**Pull request by avh4**

### CODE REVIEW

File: lib/rails_edge_test/dsl/edge.rb [hunk 1]
```diff
@@ -74,6 +74,9 @@ def produce_json(ivar: nil)
       ActiveSupport::JSON::Encoding.escape_html_entities_in_json = false
       if ivar
         value = controller.send(:instance_variable_get, ivar)
+        JSON.pretty_unparse(value.as_json)
+      elsif response[1]['Content-Type']&.starts_with?('application/json')
+        value = JSON.parse(response[2].body)
         JSON.pretty_unparse(value)
       else
         response[2].body
```
Comment (jleven) [inline lib/rails_edge_test/dsl/edge.rb]: What are you imagining is the use case of this "else" case, and why are you not prettying it?

File: lib/rails_edge_test/dsl/edge.rb [hunk 2]
```diff
@@ -74,6 +74,9 @@ def produce_json(ivar: nil)
       ActiveSupport::JSON::Encoding.escape_html_entities_in_json = false
       if ivar
         value = controller.send(:instance_variable_get, ivar)
+        JSON.pretty_unparse(value.as_json)
```
Comment (jleven) [inline lib/rails_edge_test/dsl/edge.rb]: Is `as_json` defined on all objects?

File: lib/rails_edge_test/dsl/edge.rb (hunk 1 above)
Comment (avh4) [reply lib/rails_edge_test/dsl/edge.rb]: else would be when the response body is not JSON

File: lib/rails_edge_test/dsl/edge.rb (hunk 2 above)
Comment (avh4) [reply lib/rails_edge_test/dsl/edge.rb]: yes,  `as_json` is what what we being used by itself before `JSON.pretty_unparse` was added in the previous PR

File: lib/rails_edge_test/dsl/edge.rb (hunk 1 above)
Comment (avh4) [reply lib/rails_edge_test/dsl/edge.rb]: (not needed by us at the moment, but maybe you might want to use rails_edge_test with an endpoint that returns XML or CSV or HTML or something?

File: lib/rails_edge_test/dsl/edge.rb (hunk 2 above)
Comment (avh4) [reply lib/rails_edge_test/dsl/edge.rb]: oh... oops, I guess it was `to_json` before.  I'll change it to that...

Comment (avh4) [reply lib/rails_edge_test/dsl/edge.rb]: Ah, apparently to_json returns  a string, and as_json returns ruby array/hashes.  I think as_json is correct for this purpose, and is the preferred method to override as of rails 3: http://jonathanjulian.com/2010/04/rails-to_json-or-as_json/

**Pull request merged**